### DISCUSSION TITLE
Unable to resolve: connection refused (Linux Mint)

### DISCUSSION THREAD

**Discussion by elXames** [DNSCrypt/dnscrypt-proxy#1693] [Q&A]
Hi there!

I tried setting up dnscrypt-proxy on a freshly installed Linux Mint system. I did everything as described in the 'Installation on Linux' section in the Wiki. But after stopping and disabling the systemd-resolved service and configuring the relsolv.conf file, `./dnscrypt-proxy -resolve ...` only spit out this: `Unable to resolve: [read udp 127.0.0.1:44342->127.0.0.1:53: read: connection refused]`.
Also, the resolv.conf file was reset with the following code: `# Generated by NetworkManager
nameserver 127.0.0.53`.

I have no clue how to fix this...

**Comment by ianbashford** [comment]
Hi @elXames,

I don't run Mint, but I'll try and help ( in lieu of someone stepping in who knows better).

I might be tempted to set 127.0.0.53 in the dnscrypt-proxy.toml file
(I prefer to keep it simple)

The way to stop nm overwrting resolv.conf is to edit 
`/etc/NetworkManager/NetworkManager.conf`
and add a line `dns=none` to the `[main]` section

QQ:
when you were following the instructions and got to the part where you run

`./dnscrypt-proxy`

Were there any error messages?  Was that part clean?
If you leave that running in one terminal window,  and open another, what do you see if you run
`ss -lp 'sport = :domain'`

I have a mini ubuntu server - that has resolv.conf set to 127.0.0.53 rather than the expected 127.0.0.1 
Trying these can help debug a little, depending on the result from the `ss` command output and what you have in dnscrypt-proxy.toml
  
`dig www.google.com @127.0.0.53`
`dig www.google.com @127.0.0.1`
